machine.go

```diff
@@ -52,7 +52,7 @@ func (m Machine) isAlreadyRegistered() bool {
 
 // toNode converts a Machine into a Tailscale Node. includeRoutes is false for shared nodes
 // as per the expected behaviour in the official SaaS
-func (m Machine) toNode(includeRoutes bool) (*tailcfg.Node, error) {
+func (h *Headscale) toNode(m Machine, includeRoutes bool) (*tailcfg.Node, error) {
```

Comment:
We really need to stop passing the whole application into a function (or make it the parent) just because we need one field.

In this case you have access to the parts of `Machine` you need on the outside, so you can compute the hostname and do:
```suggestion
func (m *Machine) toNode(hostname string, includeRoutes bool) (*tailcfg.Node, error) {
```

or, you can pass just the dns configuration and the base name:

```suggestion
func (m *Machine) toNode(baseDomain string, dnsConfig struct..., includeRoutes bool) (*tailcfg.Node, error) {
```

If we continue with this, it will be impossible to rename or change code, or mutate objects because they are used and are available all over the place.

Reply:
Resolving this comment will make merging with master (slightly) easier.

Reply:
You are so utterly right in this one...